utils/Felt.go

```diff
@@ -87,3 +90,54 @@ func FeltArrToBigIntArr(f []*felt.Felt) []*big.Int {
 	}
 	return bigArr
 }
+
+const SHORT_LENGTH = 31
+
+// ByteArrToFelt converts ByteArray to array of Felt objects.
+//
+// Parameters:
+// - s: string/bytearray to convert
+// Returns:
+// - []*felt.Felt: the array of felt.Felt objects
+// - error: an error, if any
+func ByteArrToFelt(s string) ([]*felt.Felt, error) {
+	arr, err := splitLongString(s)
+	if err != nil {
+		return nil, err
+	}
+
+	hexarr := []string{}
+	var (
+		count uint64
+		size  uint64
+	)
+
+	for _, val := range arr {
+		if len(val) == SHORT_LENGTH {
+			count += 1
+		}
+		size = uint64(len(val))
+		hexarr = append(hexarr, hex.EncodeToString([]byte(val)))
+	}
+
+	harr, err := HexArrToFelt(hexarr)
+	if err != nil {
+		return nil, err
+	}
+
+	harr = append(harr, new(felt.Felt).SetUint64(size))
+	return append([]*felt.Felt{new(felt.Felt).SetUint64(count)}, harr...), nil
+}
+
+func splitLongString(s string) ([]string, error) {
+	exp := fmt.Sprintf(".{1,%d}", SHORT_LENGTH)
+	r, err := regexp.Compile(exp)
```

Comment:
Let's use [MustCompile](https://pkg.go.dev/regexp#MustCompile) instead since we already know it's a valid regex expression. It'll s simplify the code

Reply:
Done